.github/workflows/update.yml

```diff
@@ -11,42 +11,88 @@ jobs:
 
     runs-on: ubuntu-latest
     name: Upstream cucumber update
+    permissions:
+      contents: read
+      pull-requests: read
     steps:
       - name: Setup PHP
         uses: shivammathur/setup-php@v2
         with:
-          php-version: 8.1
+          php-version: 8.4
           coverage: none
 
-      - uses: actions/checkout@v4
+      - name: Check whether to update or create PR
+        id: check-existing-pr
+        uses: actions/github-script@v7
+        with:
+          script: |
+            async function findOpenPR({owner, repo, branchPrefix}) {
+              const iterator = github.paginate.iterator(github.rest.pulls.list, {owner, repo, state: 'open'})
+              const repoFullName = `${owner}/${repo}`
 
-      - name: Install dependencies
-        run: composer update
+              for await (const response of iterator) {
+                const matchingPR = response.data.find((pr) => (
+                  pr.head.ref.startsWith(branchPrefix)
+                  && (pr.head.repo.full_name === repoFullName)
+                ))
 
-      - name: Update cucumber tag
-        id: cucumber
-        run: bin/update_cucumber
+                if (matchingPR) {
+                  return matchingPR
+                }
+              }
+
+              // No existing PRs
+              return false
+            }
 
-      - name: Re-install dependencies
-        run: composer update
-        if: steps.cucumber.outputs.cucumber_version
+            const branchPrefix = 'cucumber-update-'
 
-      - name: Update translations
-        run: bin/update_i18n
-        if: steps.cucumber.outputs.cucumber_version
+            const pr = await findOpenPR({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              branchPrefix
+            })
 
-      - name: Find changelog
-        id: changelog
-        run: bin/cucumber_changelog ${{ steps.cucumber.outputs.cucumber_version }}
-        if: steps.cucumber.outputs.cucumber_version
+            if (pr) {
+              core.info(`Found existing PR ${pr.number} for ${pr.head.ref}`)
+              core.setOutput('exists', 'true')
+              core.setOutput('checkout_branch', pr.head.ref)
+              core.setOutput('commit_branch', pr.head.ref)
+            } else {
+              const timestamp = Math.floor(Date.now() / 1000)
+              const newBranch = `do-update-${timestamp}`
```

Comment:
This is meant to be using the `branchPrefix` variable, apologies - I'll fix that.